### PR TITLE
Format and unvaluable changes

### PR DESCRIPTION
**Pull request by babenek**
- Fixed code style for text files
- Enabled test for PyPi pre-build on main branch fetch
- Balanced github actions to reduce payload for check-out and package installation.
- 'release_test' - when the application and required packages are installed with setup.py with latest versions
- 'development_test' - all packages are installed from requirements.txt with fixed versions

When the PR will be approved - required checks have to be updated.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
+# This workflow performs static analysis and checks coding style
```
Comment (meanrin) [inline .github/workflows/check.yml:1]: What do you think about renaming this file to `style.yml`, `style_check.yml` or something else that would indicate that this is about codding style and not actual tests?

File: .github/workflows/pypi.yml [hunk 2]
```diff
@@ -25,10 +27,21 @@ jobs:
...
-        python setup.py sdist bdist_wheel
+      run:
         twine upload dist/*
+
+    - name: Test for creation sphinx documentations
+      if: ${{ 'release' != github.event_name }}
+      run: |
+        cd docs
+        pip install -r requirements.txt
+        make html
```
Comment (meanrin) [inline .github/workflows/pypi.yml:47]: Do we have a negative example for this test? What will happen if `sphinx` returns error?

File: credsweeper/common/keyword_checklist.txt [hunk 3]
```diff
@@ -942,4 +942,4 @@ yellow
 yield
 zeros
 .json
-.xml
\ No newline at end of file
+.xml
```
Comment (meanrin) [inline credsweeper/common/keyword_checklist.txt:945]: Maybe I'm blind but I cannot see where is a difference :D
I assume it's related to a newline but git not showing it

File: docs/source/conf.py [hunk 4]
```diff
@@ -78,12 +78,11 @@
...
 html_static_path = ['_static']
 
-
 html_theme = 'sphinx_rtd_theme'
 
 html_theme_options = {
-    'logo_only': True,
-    'navigation_depth': 3
+    'logo_only': True,  #
+    'navigation_depth': 3  #
```
Comment (meanrin) [inline docs/source/conf.py:85]: What is the purpose of these comments?

Comment (babenek) [reply docs/source/conf.py:85]: to keep the lines separately - YAPF merges them

File: .github/workflows/pypi.yml (hunk 2 above)
Comment (babenek) [reply .github/workflows/pypi.yml:47]: In previous PR i added default key -W to fail in warning case

File: .github/workflows/check.yml (hunk 1 above)
Comment (babenek) [reply .github/workflows/check.yml:1]: There are multiple check types.

Comment (babenek) [reply .github/workflows/check.yml:1]: It is a common name i suppose.

**Review by csh519 — APPROVED** [review]
Thank you for clean workflow files and formatting. :broom: 

LGTM :+1:

**Review by kmnls — APPROVED** [review]

**Review by iuriimet — APPROVED** [review]

**Pull request merged**